janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java

```diff
@@ -702,7 +702,9 @@ private SearchParams convertQuery(Condition<?> condition, final KeyInformation.S
             final String key = atom.getKey();
             KeyInformation ki = information.get(key);
             final JanusGraphPredicate janusgraphPredicate = atom.getPredicate();
-            if (value instanceof Number) {
+            if (value == null && janusgraphPredicate == Cmp.NOT_EQUAL) {
+                params.addQuery(new MatchAllDocsQuery(), BooleanClause.Occur.MUST);
```

Comment:
But document may not contain field with this key name

Reply:
Thanks for the review. I am not sure If I got you, can you elaborate a bit / give an example? Are you talking about the case when a mixed index is built, but reindexing not yet done?

Reply:
Suppose next vertex schema

```
id
name
phone
```

and query `has('phone')`

Some vertices has `phone` field, some none. But lucene index build query with MatchAllDocsQuery and ignore specific filed `phone`. All vertices will be retrieved

Reply:
thanks! I shouldn't have overlooked this scenario. I will look into it.